server/models/service-process.js

```diff
@@ -8,23 +8,18 @@ module.exports = function extendServiceProcess(ServiceProcess) {
       pInfo.wid, pInfo.pid, pInfo.ppid, pInfo.pst || pInfo.startTime);
     // XXX(sam) why .pst || .startTime?
 
-    var forkedProcess = {
-      pid: pInfo.pid,
-      parentPid: pInfo.ppid,
-      workerId: pInfo.wid,
-      serviceInstanceId: instanceId,
-      startTime: new Date(pInfo.pst || pInfo.startTime),
-    };
-    var filter = {where: forkedProcess};
-
-    ServiceProcess.findOrCreate(filter, forkedProcess, function(err, proc) {
-      debug('upsert Process: %j', err || proc, pInfo);
-      if (err) return callback(err);
-      proc.updateAttributes({
-        stopReason: '',
-        stopTime: null,
-      }, callback);
-    });
+    _findOrCreateProcess(
+      instanceId, +pInfo.wid, +pInfo.pid, +pInfo.pst || +pInfo.startTime,
+      function(err, proc) {
```

Comment:
nit: if you name this function, you won't have to wrap line 11, and you will be able to outdent the named function to its previous level, and since its body hasn't change, there won't be any diff left here, so it will be really obvious that this is a refactor to delete the direct call to .findOrCreate, and replace with a call to _findOrCreateProcess.